services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchDeleteSpec.groovy

```diff
@@ -1,281 +1,194 @@
-                .getAllSwitchLinks(switchId)
-                .findAll { it.state == IslChangeType.DISCOVERED }
+        given: "An active switch with active ISLs"
+        def sw = topology.getActiveSwitches()[0]
+        def swIsls = topology.getRelatedIsls(sw)
 
-        when: "Try to force delete active switch with ISLs"
-        def response = northbound.deleteSwitch(switchId, true)
+        when: "Try to force delete the switch"
+        def response = northbound.deleteSwitch(sw.dpId, true)
 
-        then: "Check that switch is actually deleted"
+        then: "The switch is actually deleted"
         response.deleted
         Wrappers.wait(WAIT_OFFSET) {
-            !northbound.allSwitches.any { it.switchId == switchId }
-        }
-
-        and: "Cleanup: Restore the switch"
-        lockKeeper.knockoutSwitch(switchId)
-        lockKeeper.reviveSwitch(switchId)
-        Wrappers.wait(WAIT_OFFSET) {
-            northbound.activeSwitches.any { it.switchId == switchId }
-        }
+            assert !northbound.allSwitches.any { it.switchId == sw.dpId }
 
-        downLinkPorts(expectedSwitchLinks)
-        upLinkPorts(expectedSwitchLinks)
-        Wrappers.wait(antiflapCooldown + discoveryInterval + WAIT_OFFSET * 2) {
-            areLinksWereRestored(switchId, expectedSwitchLinks)
+            def links = northbound.getAllLinks()
+            swIsls.each {
+                assert !islUtils.getIslInfo(links, it)
+                assert !islUtils.getIslInfo(links, islUtils.reverseIsl(it))
+            }
         }
-    }
 
-    def downLinkPorts(List<IslInfoData> links) {
-        links.collectMany { link -> [link.source, link.destination] }
-                .each { northbound.portDown(it.switchId, it.portNo) }
-    }
-
-    def upLinkPorts(List<IslInfoData> links) {
-        links.collectMany { link -> [link.source, link.destination] }
-                .each { northbound.portUp(it.switchId, it.portNo) }
-    }
-
-    boolean areLinksWereRestored(SwitchId switchId, List<IslInfoData> expectedSwitchLinks) {
-        def actualSwitchLinks = northbound.getAllSwitchLinks(switchId)
-
-        return expectedSwitchLinks.every {
-            def expectedIsl = it
-            def actualIsl = actualSwitchLinks.findAll {
-                it.source == expectedIsl.source && it.destination == expectedIsl.destination
-            }
+        and: "Cleanup: restore the switch and ISLs"
+        // restore switch
+        lockKeeper.knockoutSwitch(sw.dpId)
+        lockKeeper.reviveSwitch(sw.dpId)
+        Wrappers.wait(WAIT_OFFSET) { assert northbound.activeSwitches.any { it.switchId == sw.dpId } }
 
-            actualIsl.size() == 1 && actualIsl.every { it.state == IslChangeType.DISCOVERED }
+        // restore ISLs
+        def allSwIsls = swIsls.collectMany { [it, islUtils.reverseIsl(it)] }
+        allSwIsls.each { northbound.portDown(it.srcSwitch.dpId, it.srcPort) }
+        TimeUnit.SECONDS.sleep(antiflapCooldown + 1)
```

Comment:
I think this is an excess 1s sleep. bare `antiflapCooldown` will do just fine even if there is 1 isl(the api call itself takes additional ~0.3-0.5s), but especially given there are N isls and you iterate through them twice, so the portUp for isl1 will be called only after portDowns for all isls are called

Reply:
Fixed.